### PR TITLE
CT-2109 [CT-2132] denormalise extend for pit deadline flag

### PR DESCRIPTION
**Pull request by mseedat-moj**
### Summary
Copies the implementation of a property/flag  (`Case::SAR#deadline_extended?` in #661) to indicate that a particular case has a current Public Interest Test (PIT) extension. This replaces the need to look-up the `case_transitions` table.

### Ticket
https://dsdmoj.atlassian.net/browse/CT-2132

### Issues

`CaseExtendForPITService` and `CaseRemovePITExtensionService` in their respective `call` methods sets an `acting_team`. 

Current implementation:

```
acting_team: BusinessUnit.dacu_bmt
```

Should be:

```
acting_team: @user.team_for_case(@case)
```

However, setting an appropriate team for the case causes a number of tests to fail including but not limited to:

```
rspec ./spec/features/cases/foi/case_extend_for_pit_spec.rb:13 # Applying a PIT extension to a case manager applies the extension
rspec ./spec/features/user_journeys/foi/compliance_review_trigger_e2e_spec.rb:41 # FOI compliance review case that requires clearance end-to-end journey
rspec ./spec/features/user_journeys/foi/trigger_e2e_spec.rb:43 # FOI case that does not require clearance end-to-end journey
rspec ./spec/features/user_journeys/overturned_foi/ovt_foi_non_trigger_e2e_spec.rb:12 # ICO Overturned FOI case end-to-end journey
rspec ./spec/features/user_journeys/overturned_foi/ovt_foi_trigger_e2e_spec.rb:45 # FOI case that requires clearance end-to-end journey

rspec ./spec/services/case_extend_for_pit_service_spec.rb:24 # CaseExtendForPITService#call calls extend_for_pit on the case state machine
rspec ./spec/services/case_extend_for_pit_service_spec.rb:35 # CaseExtendForPITService#call sets the external deadline on the case
rspec ./spec/services/case_extend_for_pit_service_spec.rb:41 # CaseExtendForPITService#call sets result to :ok and returns same
rspec ./spec/services/case_search_service_spec.rb:229 # CaseSearchService#call filtering on search results applying filter on search results search and filters have already been used by this user retrieves the existing SearchQuery

rspec ./spec/views/users/show_html_slim_spec.rb:31 # users/show.html.slim has correct data in case rows
```

This may indicate a more general issue regarding assignment of teams. A new ticket has been created:

https://dsdmoj.atlassian.net/browse/CT-2130

### CODE REVIEW

File: spec/views/users/show_html_slim_spec.rb [hunk 1]
```diff
@@ -31,14 +31,14 @@
...
-    row = users_show_page.case_list.first
+    row = users_show_page.case_list.last
     expect(row.number).to have_text(kase_1.number)
     expect(row.type).to have_text 'FOI'
     expect(row.request_detail).to have_text kase_1.subject
     expect(row.external_deadline).to have_text I18n.l(kase_1.external_deadline)
     expect(row.status).to have_text 'Draft in progress'
 
-    row = users_show_page.case_list.last
+    row = users_show_page.case_list.first
```
Comment (mseedat-moj) [inline spec/views/users/show_html_slim_spec.rb]: Why the change in ordering!? Odd!

File: app/services/case_remove_pit_extension_service.rb [hunk 2]
```diff
@@ -7,12 +7,16 @@ def initialize(user, kase)
...
 
+  # TODO (Mohammed Seedat): Setting acting_team: @user.team_for_case(@case)
+  # causes many existing test failures
   def call
     ActiveRecord::Base.transaction do
-      @case.state_machine.remove_pit_extension!(acting_user: @user,
-                                                acting_team: BusinessUnit.dacu_bmt)
-      @case.update external_deadline: (find_original_final_deadline)
-      @case.reload
+      @case.state_machine.remove_pit_extension!(
```
Comment (beccapearce) [inline app/services/case_remove_pit_extension_service.rb]: Would be good to move the state machine change into the model method remove_pit_deadline!

File: app/services/case_extend_for_pit_service.rb [hunk 3]
```diff
@@ -9,15 +9,20 @@ def initialize(user, kase, extension_deadline, reason)
...
   def call
     ActiveRecord::Base.transaction do
       if validate_params
-        @case.state_machine.extend_for_pit!(acting_user: @user,
-                                            acting_team: BusinessUnit.dacu_bmt,
-                                            final_deadline: @extension_deadline,
-                                            original_final_deadline: @case.external_deadline,
-                                            message: @reason)
-        @case.reload.update! external_deadline: @extension_deadline
+        @case.state_machine.extend_for_pit!(
```
[the remark this answers is not in the document]
Comment (mseedat-moj) [reply app/services/case_extend_for_pit_service.rb]: Decided to leave in place pending further technical discussions about current system design

File: app/services/case_remove_pit_extension_service.rb (hunk 2 above)
Comment (mseedat-moj) [reply app/services/case_remove_pit_extension_service.rb]: Decided to leave in place pending further technical discussions about current system design

File: spec/views/users/show_html_slim_spec.rb (hunk 1 above)
Comment (mseedat-moj) [reply spec/views/users/show_html_slim_spec.rb]: The update factory `trait :extended_for_pit` and `trait :pit_extension_removed` explicitly set a new PIT deadline extension. The ordering is thus changed. Not entirely sure of the process but that seems to be the cause.

**Review by beccapearce — APPROVED** [review]
